### ISSUE TITLE
Run / Connect to chainlink node

### ISSUE THREAD

**Issue by tamriq** [sorrentum/sorrentum#172]
It appeared that we need to do a couple more infra actions to connect our contracts to chainlink. The next step for us is to **connect to existing chainlink node** / or to **run our own node**. We need to choose which way we want to follow:
1) Running our own Chainlink node provides control, privacy, and customization, allowing us to manage configurations, handle sensitive data, and optimize performance. However, it requires more maintenance, cost, time and effort.
What we need to do to run our own chainlink node: 
- https://docs.chain.link/chainlink-nodes/v1/running-a-chainlink-node/
- https://www.leewayhertz.com/set-up-chainlink-node/
- https://www.quicknode.com/guides/infrastructure/node-setup/how-to-setup-a-chainlink-node/

TL;DR: we need to host one more POSTGRES database + host&run chainlink docker container on our server to make it always accessible https://hub.docker.com/r/smartcontract/chainlink

2) Using an existing Chainlink node offers ease of use, an established reputation, and shared costs. It reduces infrastructure responsibilities. On the downside, it limits control, raises privacy concerns, and creates dependency on a third-party's availability and performance.
To find a Chainlink node to connect to, we can use the Chainlink Market (https://market.link/). 

Not sure rn which option to choose, but it seems like running our own node makes sense in our case since we're going to delegate more functionality to off-chain computations. Will add more context to this issue today.

FYI @jsmerix @gpsaggese

**Comment by tamriq** [comment]
The steps we need to do to run our own chainlink node (instruction summary from https://docs.chain.link/chainlink-nodes/v1/running-a-chainlink-node):
1) Run a docker container with PostgresSQL 
2) On our server create a directory for chainlink, e.g. `chainlink-goerli`, create config `config.toml`, and `secrets.toml` that will store a keys to DB
3) Run chainlink docker container to start the node
 https://hub.docker.com/r/smartcontract/chainlink/tags

@jsmerix It looks like we need to layer all this chainlink docker stuff on our docker environment...I'm not sure how to approach this, so it would be useful if u give any thoughts on this matter.

**Comment by jsmerix** [comment]
> 1. Run a docker container with PostgresSQL

I suppose we can reuse the RDS instance we have setup in #74 for test/dev purposes. For production deployment we can discuss if we want/need to have a separate instance.

> 2. On our server create a directory for chainlink, e.g. chainlink-goerli, create config config.toml, and secrets.toml that will store a keys to DB

> 3. Run chainlink docker container to start the node

- I think we can move fairly quickly with a skateboard approach, spinning up a node for testing/development should not take too long. Of course having a production level deployment (with high availability/fault tolereance) would require a bit more effort.
- I will try to use ECS for test deployment. If the ECS approach fails, we can stick to a classic EC2 + docker approach.

**Comment by tamriq** [comment]
> I suppose we can reuse the RDS instance we have setup in https://github.com/sorrentum/sorrentum/issues/74 for test/dev purposes.

Sounds good! 

> I think we can move fairly quickly with a skateboard approach, spinning up a node for testing/development should not take too long.

For now the set-up for testing/development would be more than enough!

**Comment by tamriq** [comment]
I can provide u the additional information for config.toml file, namely WSURL and HTTPURL for this part of config, ping me when u need it.
```
[[EVM.Nodes]]
Name = 'Goerli'
WSURL = 'wss://CHANGE_ME'
HTTPURL = 'https://CHANGE_ME'  
```

**Comment by jsmerix** [comment]
@tamriq one architectural question, if we use our own chainlink node, does that imply we could run the TWAP/VWAP adapter (and other adapters that might be needed later on) privately (no public iP attached), or do they still need to be accessed publicly?

**Comment by tamriq** [comment]
@jsmerix As fas as I understand, in this case it's enough to have an adapter on the same private network as the node. It can be on the same server or on a separate server within the same network.
